components/claim/components/email_to_oauth.tsx

```diff
@@ -12,26 +11,34 @@ import {t} from 'utils/i18n.jsx';
 import LoginMfa from 'components/login/login_mfa.jsx';
 import LocalizedInput from 'components/localized_input/localized_input';
 
-export default class EmailToOAuth extends React.PureComponent {
-    static propTypes = {
-        newType: PropTypes.string,
-        email: PropTypes.string,
-        siteName: PropTypes.string,
-    };
+export type State = {
+    showMfa?: boolean;
+    password?: string;
+    error?: string;
+};
 
-    constructor(props) {
-        super(props);
+type Props = {
+    email: string;
+    newType: string;
+    siteName: string;
+}
+export default class EmailToOAuth extends React.PureComponent <Props, State> {
+    private passwordInput: React.RefObject<HTMLInputElement>;
 
-        this.state = {showMfa: false, password: ''};
+    constructor(props: Props) {
+        super(props);
 
         this.passwordInput = React.createRef();
     }
 
-    preSubmit = (e) => {
+    preSubmit = (e: React.SyntheticEvent) => {
         e.preventDefault();
-        var state = {};
 
-        var password = this.passwordInput.current.value;
+        const state = {
+            error: '',
+        };
+
+        const password = this.passwordInput.current && this.passwordInput.current.value;
```

Comment:
This pattern can be simplified using the `?.` operator to
```javascript
const password = this.passwordInput.current?.value;
```
wherever we're using it in these components